pkg/cmd/admin/migrate/storage/storage.go

```diff
@@ -0,0 +1,181 @@
+
+	"github.com/openshift/origin/pkg/cmd/admin/migrate"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+)
+
+const (
+	internalMigrateStorageLong = `
+Migrate internal object storage via update
+
+This command invokes an update operation on every API object reachable by the caller. This forces
+the server to write to the underlying storage if the object representation has changed. Use this
+command to ensure that the most recent storage changes have been applied to all objects (storage
+version, storage encoding, any newer object defaults).
+
+To operate on a subset of resources, use the --include flag. If you encounter errors during a run
+the command will output a list of resources that received errors, which you can then re-run the
+command on. You may also specify --from-key and --to-key to restrict the set of resource names
+to operate on (key is NAMESPACE/NAME for resources in namespaces or NAME for cluster scoped
+resources). --from-key is inclusive if specified, while --to-key is exclusive.
+
+By default, events are not migrated since they expire within a very short period of time. If you
+have significantly increased the expiration time of events, run a migration with --include=events
+
+WARNING: This is a slow command and will put significant load on an API server. It may also
+  result in significant intra-cluster traffic.`
+
+	internalMigrateStorageExample = `  # Perform a dry-run of updating all objects
+  %[1]s
+
+  # To actually perform the update, the confirm flag must be appended
+  %[1]s --confirm
+
+  # Only migrate pods
+  %[1]s --include=pods --confirm
+
+  # Only pods that are in namespaces starting with "bar"
+  %[1]s --include=pods --confirm --from-key=bar/ --to-key=bar/\xFF
+`
+)
+
+type MigrateAPIStorageOptions struct {
+	migrate.ResourceOptions
+}
+
+// NewCmdMigrateAPIStorage implements a MigrateStorage command
+func NewCmdMigrateAPIStorage(name, fullName string, f *clientcmd.Factory, in io.Reader, out, errout io.Writer) *cobra.Command {
+	options := &MigrateAPIStorageOptions{
+		ResourceOptions: migrate.ResourceOptions{
+			In:     in,
+			Out:    out,
+			ErrOut: errout,
+
+			Include: []string{"*"},
+			DefaultExcludes: []unversioned.GroupResource{
+				{Resource: "appliedclusterresourcequotas"},
+				{Resource: "bindings"},
+				{Resource: "deploymentconfigrollbacks"},
+				{Resource: "events"},
+				{Resource: "imagestreamimages"}, {Resource: "imagestreamtags"}, {Resource: "imagestreammappings"}, {Resource: "imagestreamimports"},
+				{Resource: "projectrequests"}, {Resource: "projects"},
+				{Resource: "componentstatuses"},
+				{Resource: "clusterrolebindings"}, {Resource: "rolebindings"},
+				{Resource: "clusterroles"}, {Resource: "roles"},
+				{Resource: "resourceaccessreviews"}, {Resource: "localresourceaccessreviews"}, {Resource: "subjectaccessreviews"},
+				{Resource: "selfsubjectrulesreviews"}, {Resource: "localsubjectaccessreviews"},
+				{Resource: "replicationcontrollerdummies.extensions"},
```

Comment:
It's added in extensions/v1beta1 due to questionable decisions around the
fake scale resource.  It has to be excluded.

On Wed, Jul 27, 2016 at 4:48 PM, Maciej Szulik <notifications@github.com>
wrote:

> In pkg/cmd/admin/migrate/storage/storage.go
> <https://github.com/openshift/origin/pull/9240#discussion_r72520506>:
>
> > +			ErrOut: errout,
> > +
> > +			Include: []string{"*"},
> > +			DefaultExcludes: []unversioned.GroupResource{
> > +				{Resource: "appliedclusterresourcequotas"},
> > +				{Resource: "bindings"},
> > +				{Resource: "deploymentconfigrollbacks"},
> > +				{Resource: "events"},
> > +				{Resource: "imagestreamimages"}, {Resource: "imagestreamtags"}, {Resource: "imagestreammappings"}, {Resource: "imagestreamimports"},
> > +				{Resource: "projectrequests"}, {Resource: "projects"},
> > +				{Resource: "componentstatuses"},
> > +				{Resource: "clusterrolebindings"}, {Resource: "rolebindings"},
> > +				{Resource: "clusterroles"}, {Resource: "roles"},
> > +				{Resource: "resourceaccessreviews"}, {Resource: "localresourceaccessreviews"}, {Resource: "subjectaccessreviews"},
> > +				{Resource: "selfsubjectrulesreviews"}, {Resource: "localsubjectaccessreviews"},
> > +				{Resource: "replicationcontrollerdummies.extensions"},
>
> Where or when that was used. My grep foo is failing me.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9240/files/1ddd7d050a21f14d07b8d4198df6e7eec9cf8475#r72520506>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_py-DyRtoqpFL2xC6snVyoRKU3775ks5qZ8QvgaJpZM4IxnZA>
> .
>